design-toolkit/components/src/components/pagination/pagination.test.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { render, screen } from '@testing-library/react';
+import { describe, expect, it } from 'vitest';
+import { Pagination } from './index';
+import type { BasePaginationProps } from './types';
+
+function setup({ currentPage, pageCount }: Partial<BasePaginationProps> = {}) {
+  render(<Pagination currentPage={currentPage} pageCount={pageCount} />);
+}
+
+describe('Pagination', () => {
+  it('should render', () => {
+    setup({ currentPage: 1, pageCount: 5 });
+    expect(screen.getByText('1')).toBeInTheDocument();
+  });
```

Comment:
not sure how robust the other component tests are for the toolkit since we have storybook integration testing, and if we want to beef this up, consider adding tests for:
- page navigation
- edge cases (first/last page, single page)
- disabled state logic
- range calculation function